grisera_api/time_series/transformation/multidimensional/TimeSeriesTransformationMultidimensional.py

```diff
@@ -0,0 +1,71 @@
+from typing import List
+
+from time_series.helpers import get_node_property
+from time_series.time_series_model import TimeSeriesOut, Type, TimeSeriesMultidimensionalOut
+
+
+class TimeSeriesTransformationMultidimensional:
+    """
+    Class with logic of time series multidimensional transformation
+
+    """
+
+    def transform(self, time_series: List[TimeSeriesOut]):
+        """
+        Transform time series data.
+
+        Get signal values lists grouped by timestamp values.
+        This transformation will ignore all signal values which timestamps will not be equal.
+
+        Args:
+            time_series (List[TimeSeriesOut]): Time series to be transformed
+
+        Returns:
+            New time series object
+        """
+        assert len(time_series) > 0, "Number of time series should be at least 1 for multidimensional transformation"
+        for i in range(1, len(time_series)):
+            assert time_series[0].type == time_series[i].type, "Time series types should be equal"
+
+        new_signal_values = []
+        current_signal_value_rest_indexes = [0] * len(time_series)
+        timestamp_label = "timestamp" if time_series[0].type == Type.timestamp else "start_timestamp"
+        # Iterate over all X signal values
+        for current_signal_value_x_index in range(len(time_series[0].signal_values)):
```

Comment:
Probably I don't see something... Could you explain why you are not using:
```python
for current_signal_value_x in time_series[0].signal_values:
```

Reply:
I fixed that